File: java/secret-handshake/src/main/java/HandshakeCalculator.java
import java.util.Collections;
import java.util.List;
import java.util.ArrayList;


class HandshakeCalculator {

    List<Signal> calculateHandshake(int number) {
        var signals = new ArrayList<Signal>();
        if( (number & 0b00000001) != 0){
            signals.add(Signal.WINK);
        }
        if( (number & 0b00000010) != 0 ) {
            signals.add(Signal.DOUBLE_BLINK);
        }
        if( (number & 0b00000100) != 0){
            signals.add(Signal.CLOSE_YOUR_EYES);
        }
        if( (number & 0b00001000) != 0 ){
            signals.add(Signal.JUMP);
        }
        if( (number & 0b0010000) != 0) {
            Collections.reverse(signals);
        }
        return signals;
    }

}

File: kotlin/secret-handshake/src/main/kotlin/HandshakeCalculator.kt
object HandshakeCalculator {

    private infix fun Int.isBitSet(bit: Int): Boolean = (this shr (bit - 1)).and(0b00000001) != 0

    fun calculateHandshake(number: Int): List<Signal> {
        return mutableListOf<Signal>().apply {
            if (number.isBitSet(1)) {
                add(Signal.WINK)
            }
            if( number.isBitSet(2)) {
                add(Signal.DOUBLE_BLINK)
            }
            if( number.isBitSet(3)){
                add(Signal.CLOSE_YOUR_EYES)
            }
            if( number.isBitSet(4)){
                add(Signal.JUMP)
            }
            if( number.isBitSet(5)){
                reverse()
            }
        }
    }
}
